### PR TITLE
Deck builder edit name + show user

### PR DESCRIPTION
**Pull request by lily-pytel**
PR 2 for deck builder refactor. This time:

- The dedicated place for editing the deck name was removed. Now merged with the name display.
- Added a "by []" that either shows guest and invites the user to log in or shows their user name

Again, no new tests because this ain't over yet. However, this branch and the parent branch could be deployed at any time since they're just a refactor of the existing state. Once we start messing more with the page, it won't be the case

### CODE REVIEW

File: next/components/edit-deck-name.js [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
-      placeholder="Enter a name..."
-    />
+    <div className="deck-name-container">
+      <style jsx>{`
+        input {
+          width: 100%;
+          background-color: transparent;
+          border: ${theme.inputBorder};
+          border-radius: 10px;
+          font-size: 20px;
```
Comment (alexmiddeleer) [inline next/components/edit-deck-name.js]: Not your fault of course, and we've all been doing it, but I hate having font sizes scattered about. Maybe next project I'll insist on a set of agreed standard label styles.

File: next/components/deck-builder-user.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        }
+        .user-link {
+          font-size: inherit;
+        }
+      `}</style>
+      <div>
+        by{' '}
+        <Link href={user ? '/account' : '/auth/google'}>
+          <a data-cy="userLink" className="user-link">
+            {user?.username || 'Guest'}
```
[the remark this answers is not in the document]
Comment (lily-pytel) [reply next/components/deck-builder-user.js]: @alexmiddeleer So many times I wrote `&&` and then was like `NO! I don't have to live like this anymore!!!`

File: next/components/deck-builder-user.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          text-decoration: none;
+          content: '\u25b6';
+          margin-right: 3px;
+          font-size: 10px;
+        }
+        .login-link {
+          font-size: 14px;
+        }
+        .user-link {
+          font-size: inherit;
```
[the remark this answers is not in the document]
Comment (lily-pytel) [reply next/components/deck-builder-user.js]: I swear there was at some point 🤦‍♂ but not anymore so removed

File: next/components/deck-builder-user.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          margin: 10px 0;
+        }
+        .login-link,
+        .user-link {
+          text-decoration: none;
+          color: ${theme.fontColorAccent};
+        }
+        .login-link:before {
+          text-decoration: none;
+          content: '\u25b6';
```
[the remark this answers is not in the document]
Comment (lily-pytel) [reply next/components/deck-builder-user.js]: image

Done!

File: next/components/edit-deck-name.js (hunk 1 above)
Comment (lily-pytel) [reply next/components/edit-deck-name.js]: It doesn't help that the template has so many font sizes floating around.

Comment (lily-pytel) [reply next/components/edit-deck-name.js]: Totally agree otherwise.

**Pull request merged**